src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java

```diff
@@ -82,6 +92,26 @@
 
    private Pointer posix_spawn_file_actions;
 
+   // Launches threads under which changes to cwd do not affect the cwd of the process.
+   public static class LinuxCwdChangeableThreadFactory implements ThreadFactory {
+      private final AtomicInteger threadCount = new AtomicInteger(0);
+
+      @Override
+      public Thread newThread(final Runnable r) {
+         Runnable unshareCwdThenSpawn = new Runnable() {
+            @Override
+            public void run() {
+              int rc = LinuxLibC.unshare(LinuxLibC.CLONE_FS);
+              checkReturnCode(rc, "unshare(CLONE_FS) failed");
```

Comment:
That's a great question. If `ThreadFactory.newThread()` throws, it bubbles up to whomever called `ThreadPoolExecutor.execute()`.

http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/util/concurrent/ThreadPoolExecutor.java#ThreadPoolExecutor.addWorkerFailed%28java.util.concurrent.ThreadPoolExecutor.Worker%29

I'd expect this to basically be a fatal. This system call shouldn't fail unless there's some serious kernel corruption going on.